core/storage/exploration/gae_models_test.py

```diff
@@ -0,0 +1,347 @@
+    """Test the ExplorationRightsModel class."""
+
+    def test_save(self):
+        exploration_models.ExplorationRightsModel(
+            id='id_0',
+            owner_ids=['owner_id'],
+            editor_ids=['editor_id'],
+            translator_ids=['translator_id'],
+            viewer_ids=['viewer_id'],
+            community_owned=False,
+            status=constants.ACTIVITY_STATUS_PUBLIC,
+            viewable_if_private=False,
+            first_published_msec=0.0
+        ).save(
+            'cid', 'Created new exploration right',
+            [{'cmd': rights_manager.CMD_CREATE_NEW}])
+        saved_model = exploration_models.ExplorationRightsModel.get('id_0')
+        self.assertEqual(saved_model.id, 'id_0')
+        self.assertEqual(saved_model.owner_ids, ['owner_id'])
+        self.assertEqual(saved_model.translator_ids, ['translator_id'])
+        self.assertEqual(saved_model.viewer_ids, ['viewer_id'])
+
+
+class ExplorationCommitLogEntryModelUnitTest(test_utils.GenericTestBase):
+    """Test the ExplorationCommitLogEntryModel class."""
+
+    def test_get_all_non_private_commits(self):
+        private_commit = (
+            exploration_models.ExplorationCommitLogEntryModel.create(
+                'a', 1, 'commiter_id', 'username', 'msg',
+                'create', [{}],
+                constants.ACTIVITY_STATUS_PRIVATE, False))
+        public_commit = (
+            exploration_models.ExplorationCommitLogEntryModel.create(
+                'b', 1, 'commiter_id', 'username', 'msg',
+                'create', [{}],
+                constants.ACTIVITY_STATUS_PUBLIC, False))
+        private_commit.exploration_id = 'a'
+        public_commit.exploration_id = 'b'
+        private_commit.put()
+        public_commit.put()
+        results, _, more = (
+            exploration_models.ExplorationCommitLogEntryModel
+            .get_all_non_private_commits(2, None, None))
+        self.assertFalse(more)
+        self.assertEquals(len(results), 1)
+
+    def test_get_all_exploration_commits(self):
+        commit_v1 = (
+            exploration_models.ExplorationCommitLogEntryModel.create(
+                'a', 1, 'commiter_id', 'username', 'msg',
+                'create', [{}],
+                constants.ACTIVITY_STATUS_PRIVATE, False))
+        commit_v2 = (
+            exploration_models.ExplorationCommitLogEntryModel.create(
+                'a', 2, 'commiter_id', 'username', 'msg',
+                'edit', [{}],
+                constants.ACTIVITY_STATUS_PRIVATE, False))
+        commit_v1.exploration_id = 'a'
+        commit_v2.exploration_id = 'a'
+        commit_v1.put()
+        commit_v2.put()
+
+        results = (
+            exploration_models.ExplorationCommitLogEntryModel
+            .get_all_exploration_commits('a', 1))
+        self.assertEquals(results, [commit_v1])
+        results = (
+            exploration_models.ExplorationCommitLogEntryModel
+            .get_all_exploration_commits('a', 2))
```

Comment:
Actually, sorry about this, but while checking the return value of get_all_exploration_commits(), I just realized that it is not used at all in the codebase.

Could we delete it instead? Sorry for not realizing this earlier.

Reply:
No worries, do you mean from just the test or from both the test and the actual implementation?

Reply:
Both of them -- we shouldn't keep dead code around. Thanks!

Reply:
Done